apps/files/src/components/FilesTopBar.vue

```diff
@@ -29,11 +29,11 @@
         <oc-menu v-if="this.canUpload">
           <oc-button id="new-file-menu-btn" text="+ New" slot="activator" variation="primary"></oc-button>
           <!-- TODO: replace with oc-list elements-->
-          <ul class="uk-nav uk-dropdown-nav uk-nav-default">
+          <template slot="subnav">
             <file-upload :url='url' :headers="headers" @success="onFileSuccess" @error="onFileError" @progress="onFileProgress"></file-upload>
-            <li @click="createFolder = true" id="new-folder-btn"><a href="#"><oc-icon name="create_new_folder"/><translate>Create new folder ...</translate></a></li>
-            <li @click="createFile = true" id="new-file-btn"><a href="#"><oc-icon name="save"/><translate>Create new file ...</translate></a></li>
-          </ul>
+            <oc-menu-item @click="createFolder = true" id="new-folder-btn" icon="create_new_folder" content="Create a new folder"></oc-menu-item>
+            <oc-menu-item @click="createFile = true" id="new-file-btn" icon="save" content="Create a new file"></oc-menu-item>
```

Comment:
translate is missing

Reply:
I moved translate into ods via <template v-translate>. Otherwise I would have to edit the whole components since It was then overriding the condition with icon thus not aligning properly. I will open PR for that now so you can have a look.

Reply:
PR opened - https://github.com/owncloud/owncloud-design-system/pull/196

Reply:
this is how it should look like:
```js
            <oc-menu-item @click="createFolder = true" id="new-folder-btn" icon="create_new_folder"><translate>Create new file ...</translate></oc-menu-item>
            <oc-menu-item @click="createFile = true" id="new-file-btn" icon="save"><translate>Create new folder ...</translate></oc-menu-item>
```

We need to change the OcMenuItem component - let's think about using the default slot for the text and let there be a different slot for defining the whole inner template.

Proposal for oc-menu-item:
```html
<template>
  <li @click="onClick">
    <a href="#">
      <!-- @slot This slot is set as content & will override any inner elements-->
      <slot name="inner">
        <div v-if="icon" class="uk-flex uk-flex-middle">
          <oc-icon :name="icon" class="uk-margin-small-right" />
          <!-- @slot This default slot is set as text of the menu items -->
          <slot name="default" />
        </div>
        <template v-else>
          <slot name="default" />
        </template>
      </slot>
    </a>
  </li>
</template>
<script>
/**
 * Create a menu item.
 */
export default {
  name: "oc-menu-item",
  status: "prototype",
  release: "0.0.1",
  props: {
    /**
     * The icon to display.
     */
    icon: {
      type: String,
      required: false,
      default: "",
    },
  },
  methods: {
    onClick(val) {
      /**
       * Click event
       * @event click
       * @type {event}
       */
      this.$emit("click", val)
    },
  },
}
</script>

<docs>
```jsx
<template>
    <oc-menu buttonText="open oc-menu">
      <template slot="subnav">
        <oc-menu-item icon="create_new_folder">New folder …</oc-menu-item>
        <oc-menu-item>Another action</oc-menu-item>
        <oc-menu-item><pre slot="inner">Custom item elements</pre></oc-menu-item>
      </template>
    </oc-menu>
  </template>
  <script>
    export default {
      methods: {
        onClick(val) {
          alert(val)
        }
      }
    }
  </script>
```
</docs>

```

Reply:
feel free to grab code form here - https://github.com/owncloud/owncloud-design-system/pull/199/commits/2e2e02e1fb2f5781e1b41a42e51090581c39dd18